src/OneExplorer/OneExplorer.ts

```diff
@@ -465,38 +463,53 @@ input_path=${modelName}.${extName}
     };
 
     // Get the list of all the cfg files inside workspace root
-    const confs =
-        readdirSyncRecursive(this.workspaceRoot!.fsPath).filter(val => val.endsWith('.cfg'));
+    const cfgList = readdirSyncRecursive(root).filter(val => val.endsWith('.cfg'));
 
-    for (const conf of confs) {
-      const cfgObj = ConfigObj.createConfigObj(vscode.Uri.file(conf));
+    return cfgList;
+  }
+
+  /**
+   * Search corresponding .cfg files inside the workspace
+   * for the given baseModelNode
+   *
+   * @param baseModelNode a Node of the base model
+   */
+  private searchConfig(baseModelNode: Node) {
+    console.assert(baseModelNode.type === NodeType.baseModel);
+
+    const cfgList = this.getCfgList();
+
+    for (const cfg of cfgList) {
+      const cfgObj = ConfigObj.createConfigObj(vscode.Uri.file(cfg));
 
       if (!cfgObj) {
-        Logger.info('OneExplorer', `Failed to open file ${conf}`);
+        Logger.info('OneExplorer', `Failed to open file ${cfg}`);
         continue;
       }
 
       if (!cfgObj.isChildOf(baseModelNode.path)) {
         continue;
       }
 
-      const {baseModels, derivedModels} = cfgObj.obj;
-
-      const pairNode = new Node(NodeType.config, [], vscode.Uri.file(conf));
-      Logger.debug('OneExplorer', `DerivedModels : ${derivedModels}`);
-
-      derivedModels ?.forEach(derivedModel => {
-                       // Display only the existing node
-                       const realPath = RealPath.createRealPath(derivedModel.path);
-                       if (realPath) {
-                         pairNode.childNodes.push(new Node(
-                             NodeType.derivedModel, [], vscode.Uri.file(realPath.absPath)));
-                       }
-                     });
+      const pairNode = this.createConfigNode(cfg, cfgObj.getDerivedModelsExists);
 
       baseModelNode.childNodes.push(pairNode);
     }
   }
+
+  private createConfigNode(conf: string, derivedModels: Artifact[]): Node {
+    const pairNode = new Node(NodeType.config, [], vscode.Uri.file(conf));
+
+    Logger.debug('OneExplorer', `DerivedModels : ${derivedModels}`);
+
+    derivedModels.forEach(derivedModel => {
```

Comment:
(5) Import 'Artifact' interface

Now derivedModels is an Artifact and has 'ext', 'type' or so. OneExplorer can now get those additional informations passed.